chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -442,11 +442,9 @@ case class ChainHandler(
     getBlockHeight(blockHash).flatMap {
       case None => FutureUtil.none
       case Some(blockHeight) =>
-        for {
-          tipHash <- getBestBlockHash()
-          tipHeightOpt <- getBlockHeight(tipHash)
-        } yield {
-          tipHeightOpt.map(tipHeight => tipHeight - blockHeight + 1)
+        blockHeaderDAO.chainTips.map { tips =>
+          val bestHeader = tips.maxBy(_.chainWork)
+          Some(bestHeader.height - blockHeight + 1)
```

Comment:
This doesn't seem right. We don't know if the block is apart of these various chains do we? The block we are talking about could be a part of Chain A, but not Chain B?

Reply:
I'm not sure how we'd solve that without linking headers all the way back